### DISCUSSION TITLE
Issues with Python Semantic Highlighting & Themes

### DISCUSSION THREAD

**Discussion by akthe-at** [LazyVim/LazyVim#2792] [Q&A] [answered]
Anyone have any idea why/what is different about Tokyonight and Catppuccin that are included with Lazyvim that make them work nicely with Semantic Highlighting tokens from the new Basedpyright LSP for python? I have tried many other themes that supposedly support semantic highlights but so far only these two have worked with LazyVim for myself. 

> A few of the ones that I have tried include:
> Dracula
> Gruvbox-material
> Rose-pine

**Comment by folke** [comment, marked as answer]
TokyoNight has extensive support for semantic tokens. (and catppuccing probably as well)
Most themes don't add custom support for semantic tokens.

Apart from that I actually also fixed a small thing specifically for basedpyright in tokyonight

**Reply by akthe-at** [reply]
I saw that commit and that is why I closed it because I immediately understood what was happening with all the of themes (in addition to them not leaving the lsp variable link = {} in order to defer to tree sitter for regular variables.
